server/schedule/placement/fit.go

```diff
@@ -215,28 +219,26 @@ func replaceFitPeerOpt(srcStoreID uint64, dstStore *core.StoreInfo) fitPeerOpt {
 func newFitPeer(stores []*core.StoreInfo, region *core.RegionInfo, fitPeers []*metapb.Peer, opts ...fitPeerOpt) []*fitPeer {
 	peers := make([]*fitPeer, len(fitPeers))
 	for i, p := range fitPeers {
-		peer := &fitPeer{
-			Peer:     p,
-			store:    getStoreByID(stores, p.GetStoreId()),
-			isLeader: region.GetLeader().GetId() == p.GetId(),
-		}
+		peer := fitPeerPool.Get().(*fitPeer)
+		peer.Peer = p
+		peer.store = getStoreByID(stores, p.GetStoreId())
+		peer.isLeader = region.GetLeader().GetId() == p.GetId()
 		for _, opt := range opts {
 			opt(peer)
 		}
 		peers[i] = peer
 	}
+	return peers
+}
+
+func newFitWorker(stores []*core.StoreInfo, region *core.RegionInfo, rules []*Rule) *fitWorker {
+	peers := newFitPeer(stores, region, region.GetPeers())
 	// Sort peers to keep the match result deterministic.
 	sort.Slice(peers, func(i, j int) bool {
```

Comment:
Will `Replace` rely on the sort?

Reply:
None, The replaced peer doesn't consider the selected order and the peer state will be same, so it doesn't need to sort.